### PR TITLE
Testing with Codat

### PR DESCRIPTION
**Pull request by pzaichkina**
# Description

Updates to Startup billing and new page on Testing

## Type of change

Please delete options that are not relevant.

- [X] New document(s)/updating existing
- [ ] Fixes
- [ ] Styling
- [ ] Bug fix (non-breaking change which fixes an issue)

## Reviews and merging

You are responsible for getting your PR merged. Address review comments promptly and **make sure to merge the PR when ready**.
Feel free to 'Enable automerge' - your PR will automatically merge when accepted and passing the build.

### CODE REVIEW

File: docs/introduction/testing.md [hunk 1]
```diff
@@ -1,6 +1,84 @@
...
+
+   Involve one of your real SMB customers to perform authentication via your front-end digital application flow. Then, verify that the data syncs between the relevant platform and Codat and triggers a webhook into your system to start the data pull.
+
+## Back-end integration testing
+
+When testing your integrations throughout the implementation, it is important to check you can authorize and complete the connections before you proceed to live SMB testing.
+
+We recommend using **Codat Sandbox** integrations that contain sample data generated as a tool for developer testing. When you link to these integrations, you can choose from a number of different datasets that provide varying levels of detail. 
+
+You can also test with **Xero** and **QuickBooks Online**, both of which offer demo or sandbox companies equipped with sample data. Other integrations may require signing up to multiple free trials in order to test the connections, or provide no sample data.
```
Comment (dcoplowe) [inline docs/introduction/testing.md:55]: I think this implies that no sign up is needed for Xero and QBO to get access to their sandbox/demo data.

File: docs/introduction/testing.md [hunk 2]
```diff
@@ -1,6 +1,84 @@
...
+
+:::info Testing in production instance
+In your production instance, active companies that are connected only to Codat Sandbox or QuickBooks Online Sandbox are excluded from billing.
```
Comment (dcoplowe) [inline docs/introduction/testing.md:58]: Nice

File: docs/introduction/testing.md [hunk 3]
```diff
@@ -1,6 +1,84 @@
...
+* Integration-specific settings and credentials
+* Client sync settings
+* Webhook rules and settings
+* Auth flow settings
+
+We work directly with our **Enterprise** clients to arrange test instance creation. 
+
+Start-up and Enterprise users can switch between the production client and the test instance by using the client dropdown in the [Codat Portal](https://app.codat.io/). 
+
+![](/img/introduction/0026-client-selection.png)
```
Comment (dcoplowe) [inline docs/introduction/testing.md:27]: Nit, the screenshot contains a Codat Internal instance. Can we update to one that a client would see?

File: docs/introduction/testing.md [hunk 4]
```diff
@@ -1,6 +1,84 @@
...
+
+We recommend using **Codat Sandbox** integrations that contain sample data generated as a tool for developer testing. When you link to these integrations, you can choose from a number of different datasets that provide varying levels of detail. 
+
+You can also test with **Xero** and **QuickBooks Online**, both of which offer demo or sandbox companies equipped with sample data. Other integrations may require signing up to multiple free trials in order to test the connections, or provide no sample data.
+
+:::info Testing in production instance
+In your production instance, active companies that are connected only to Codat Sandbox or QuickBooks Online Sandbox are excluded from billing.
+:::
+
+### Overview of integrations' test accounts
```
Comment (dcoplowe) [inline docs/introduction/testing.md:61]: This section is superb. It'll really help clients.

File: docs/introduction/testing.md (hunk 3 above)
Comment (pzaichkina) [reply docs/introduction/testing.md:27]: @dcoplowe the only instances I have access to are Codat Internal and one Start-up one. if you have something else where a screenshot like that can be taken, let me know

File: docs/introduction/testing.md [hunk 5]
```diff
@@ -1,6 +1,84 @@
...
+
+Start-up and Enterprise users can switch between the production client and the test instance by using the client dropdown in the [Codat Portal](https://app.codat.io/). 
+
+![](/img/introduction/0026-client-selection.png)
+
+### Pitfalls
+
+1. Your test client will have a separate set of API keys, and you may need to maintain different credentials for the integrations in the test instance. 
+
+2. To see the new client once it's been created, you will need to reauthenticate. Log out and log back in if you are using Portal, or get a new token if you are using our APIs.
```
[the remark this answers is not in the document]
Comment (pzaichkina) [reply docs/introduction/testing.md:33]: I believe so. Ben said this is what it is currently.

[the remark this answers is not in the document]
Comment (pzaichkina) [reply docs/introduction/testing.md:31]: as in, this is how it is built for now, and I am not sure if there are future plans to rework that somehow. it's not specifically called out in the spec

**Review by dcoplowe — APPROVED** [review]

**Pull request merged**